docs - embedding a chart (#78856)
metabase/metabase: docs/embedding/introduction.md

# Embedding introduction

> You can embed Metabase tables, charts, dashboards, AI chat—even Metabase's query builder—in your website or application.

Reviewer: ```suggestion
You can embed Metabase tables, charts, dashboards, AI chat---even Metabase's query builder---in your website or application.
```
<img width="1536" height="472" alt="image" src="https://github.com/user-attachments/assets/a686e18c-7d59-4e66-a64a-2945040e2de3" />
but also i think em dash is misused here

There are two ways to embed Metabase.

- [Modular embedding](#modular-embedding). Embed individual Metabase components, like questions, dashboards, and AI chat, so they seamlessly integrate with your app.
- [Full app embedding](#full-app-embedding). Embed the full Metabase app in an iframe, styled like your branding.

Most people go with modular embedding, so you can integrate Metabase components into your app.

## Modular embedding

With [modular embedding](./modular-embedding.md), you can embed individual Metabase components in your web app. You can use guest embeds for basic functionality, or use SSO to take full advantage of Metabase.

You can use two different ways to authenticate modular embeds:

- [SSO](#sso-embeds)
- [Guest](#guest-embedding)

> ### Web components or the React SDK?

Reviewer: This header is really weird if you aren't already well-versed in metabase embedding. Let's say I'm a person encountering metabase embedding for the first time, and reading this overview article to understand it. Why would I ever have a question "Web components or the React SDK?" I don't even know that those options exist yet

You can set up modular embeds two ways: with web components, or with the React SDK.

If your app uses React, go with the SDK. The web components, which you can set up in the [in-app wizard](./modular-embedding.md), are built on top of the SDK, so either option works for a React app. The SDK just gives you more control: you can build custom layouts, and [customize behavior with plugins](./sdk/plugins.md). You can always start with web components and move to the SDK if you really need that extra customization.

> Both options support SSO and guest embeds.

Reviewer: This does not explain what the webcomponent option is like at all (e.g. that you don't need any framework, it's just an html component etc). "Built on top of the SDK" mention actually accomplish the opposite of clarifying - it makes it feel like you have to use react for it anyway

# Embedding introduction

> You can embed Metabase tables, charts, dashboards, AI chat — even Metabase's query builder — in your website or application.

Author: ```suggestion
You can embed Metabase tables, charts, dashboards, AI chat---even Metabase's query builder---in your website or application.
```

If you're building a SaaS product with embedded analytics for multiple customers, you can keep customer data isolated with [Tenants](./tenants.md).

> Accounts for these embedded users in your Metabase count toward the accounts billed in your Metabase plan. But by letting your customers self-serve their data, you save time on developing bespoke charts. And you can charge _more_ for a premium analytics experience. If you plan on giving a lot of your customers self-service access to their data, you should consider an enterprise plan, with custom pricing that scales with your business.

Author: ```suggestion
Accounts for these embedded users in your Metabase count toward the [accounts billed in your Metabase plan](https://www.metabase.com/docs/latest/cloud/how-billing-works). But by letting your customers self-serve their data, you save time on developing bespoke charts. And you can charge _more_ for a premium analytics experience. If you plan on giving a lot of your customers self-service access to their data, you should consider an enterprise plan, with custom pricing that scales with your business.
```

### Guest embedding

[Guest embeds](./guest-embedding.md) are a secure way to embed charts and dashboards. Guest embedding works on all Metabase plans, including OSS and Starter.

**When to use guest embeds**: simple embedding use cases where you don't want to offer ad-hoc querying or chart drill-through. To filter data relevant to the viewer, you can use guest embeds with [locked parameters](./guest-embedding.md#locked-parameters).

### Set up modular embeds with web components or React

Whichever way you authenticate, you can set up modular embeds two ways.

> - **Web components**: a script tag plus HTML elements like `<metabase-question>`. Web components have no build step and no framework requirement, so they work in plain HTML, Vue, Svelte, Rails, or React. Metabase's [in-app wizard](./modular-embedding.md) writes the code for you.

Reviewer: Really like this paragraph, thank you!

Reviewer: maybe we can be like "... Rails, React, or any framework of your choice" (you know PAs will read this as saying they work in _only_ these cases otherwise)

| [SQL editor](../questions/native-editor/writing-sql.md)                                   | ✅  | ❌    |
| [AI chat](./sdk/ai-chat.md)                                                               | ✅  | ❌    |
| [Collection browser](./sdk/collections.md)                                                | ✅  | ❌    |
| Advanced [Tenant](./tenants.md) and [permissions](../permissions/embedding.md) management | ✅  | ❌    |
| [Advanced theming](./appearance.md)                                                       | ✅  | ❌    |
| [Usage analytics](../usage-and-performance-tools/usage-analytics.md)                      | ✅  | ❌    |
| Customize layouts and behavior with [plugins](./sdk/plugins.md)                           | ✅  | ❌    |
| [Locked filters](./guest-embedding.md#locked-parameters)\*\*\*                            | ❌  | ✅    |

\* Each embedding type allows data downloads by default, but only [Pro and Enterprise](https://www.metabase.com/pricing/) plans can disable data downloads.

\*\* Requires a [Pro and Enterprise](https://www.metabase.com/pricing/) plan for any embedding type.

> \*\*\* SSO embeds don't need locked filters. Since Metabase knows who's viewing an SSO embed, you can segregate data with [permissions](../permissions/embedding.md) instead.

Author: ```suggestion
\*\*\* SSO embeds don't need locked filters. Since Metabase knows who's viewing an SSO embed, you can segregate data with [permissions](../permissions/embedding.md) instead. There's a little more set up, but much less long-term overhead.
```

If you're building a SaaS product with embedded analytics for multiple customers, you can keep customer data isolated with [Tenants](./tenants.md).

> Creating accounts for these embedded users in your Metabase counts toward the accounts billed in your Metabase plan. But by letting your customers self-serve their data, you save time on developing bespoke charts. And you can charge _more_ for a premium analytics experience. If you plan on giving a lot of your customers self-service access to their data, you should consider an enterprise plan, with custom pricing that scales with your business.

Reviewer: ```suggestion
Accounts for these embedded users in your Metabase count toward the accounts billed in your Metabase plan. But by letting your customers self-serve their data, you save time on developing bespoke charts. And you can charge _more_ for a premium analytics experience. If you plan on giving a lot of your customers self-service access to their data, you should consider an enterprise plan, with custom pricing that scales with your business.
```

i feel like permissions are a much better selling point than self service tbh...But i also don't really think we should be trying to sell anything in docs.

| [Basic appearance customization](../configuring-metabase/appearance.md)\*\*               | ✅  | ✅    |
| Row-level data segregation                                                                | ✅  | ✅    |
| [Drill-through menus](../questions/visualizations/drill-through.md)                       | ✅  | ❌    |
| [Query builder](../questions/query-builder/editor.md)                                     | ✅  | ❌    |
| [SQL editor](../questions/native-editor/writing-sql.md)                                   | ✅  | ❌    |
| [AI chat](./sdk/ai-chat.md)                                                               | ✅  | ❌    |
| [Collection browser](./sdk/collections.md)                                                | ✅  | ❌    |
| Advanced [Tenant](./tenants.md) and [permissions](../permissions/embedding.md) management | ✅  | ❌    |
| [Advanced theming](./appearance.md)                                                       | ✅  | ❌    |
| [Usage analytics](../usage-and-performance-tools/usage-analytics.md)                      | ✅  | ❌    |
| Customize layouts and behavior with [plugins](./sdk/plugins.md)                           | ✅  | ❌    |

> **When to use public links and embeds**: One-off charts and dashboards. Admins can use public links when you just need to show someone a chart or dashboard without giving people access to your Metabase. And you don't care who sees the data; you want to make the item available to everyone.

Reviewer: But now we never mention public links or embeds, so this comes out of nowhere...

Reviewer: Oh i see they're mentioned below. Maybe this section should be removed here then?

| Customize behavior with [plugins](./sdk/plugins.md)                                                          | ✅                                   | ❌                                    | ❌                                    | ❌                                  | ❌                                     |
| [Custom visualizations](../questions/visualizations/custom.md)                                               | ✅                                   | ❌                                    | ❌                                    | ❌                                  | ❌                                     |
| AI chat                                                                                                      | ✅                                   | ✅                                    | ❌                                    | ✅                                  | ❌                                     |

> >>>>>>> master

Reviewer: ```suggestion
```

## Embedding limitations

> <<<<<<< HEAD

Reviewer: ```suggestion
```

> Currently, you can't embed [documents](../documents/introduction.md) (though you can create [public documents](./public-links.md)).

Reviewer: so like you can embed a public link in an iframe, right?

## Full app embedding

[Full app embedding](./full-app-embedding.md) allows you to embed the entire Metabase app in an iframe, and integrate Metabase SSO with your app's authentication.

## Public links and embeds

If you'd like to share your data with the good people of the internet, admins can create a [public link](./public-links.md) or embed a question or dashboard directly in your website. Public links aren't really an embedding setup—there's no authentication, and anyone with the link can see the data.

**When to use public links and embeds**: one-off charts and dashboards. Admins can use public links when you just need to show someone a chart or dashboard without giving people access to your Metabase. And you don't care who sees the data; you want to make the item available to everyone.

> =======

Reviewer: ```suggestion
```

- Currently, you can't embed [documents](../documents/introduction.md) (though you can create [public documents](./public-links.md)).
- Only the [Modular embedding SDK](./sdk/introduction.md) renders [custom visualizations](../questions/visualizations/custom.md), and only ones you allowlist with the [`allowedCustomVisualizations` prop](./sdk/config.md#custom-visualizations). In other embedding types, any card that uses a custom visualization falls back to the default visualization for the query's results.

> >>>>>>> master

Reviewer: ```suggestion
```
